app/controllers/brapi/v1/phenotypes_controller.rb

```
@@ -17,100 +17,95 @@ def search
+          observation_row["observations"] = observations[observation_unit_id]
+          observation_units << observation_row
+        end  
+        observations[observation_unit_id] << observation
       end
       
-      json_result_array = intermediate_hash.values
-      
       # pagination data returned
-      
-      result_count_object = phenotypes_queries.phenotypes_search_query(
-      params['germplasmDbIds'], params['observationVariableDbIds'], params['studyDbIds'], params['locationDbIds'],
-      params['seasonDbIds'], 
-      params['sortBy'], params['sortOrder'], page, page_size, true)
+      result_count_object = phenotypes_queries.phenotypes_search_query(query_params, count_mode: true)
       
       total_count = result_count_object.values.first[0].to_i
       total_pages = (total_count/page_size.to_f).ceil
       
       json_response = { 
         metadata: json_metadata(page_size, page, total_count, total_pages),
         result: {
-          data: json_result_array
+          data: observation_units
         }
       }
-     
       render json: json_response, except: ["id", "user_id", "created_at", "updated_at", "total_entries_count"]
-   
-    
+    else
+      render json: { reason: 'Resource not found' }, status: :not_found  # 404
     end
   end
 
 
   private
 
+  def extract_observation(observation_row)        
+    # We introduce data related to plot numbers, plants, etc
+    begin
+      observation_levels_hash = JSON.load(observation_row.delete("observation_levels_json"))
+      if !observation_levels_hash.nil?
+          observation_row["plotNumber"]= observation_levels_hash["plot"]
+          observation_row["plantNumber"]= observation_levels_hash["plant"]
+          observation_row["blockNumber"]= observation_levels_hash["block"]
+          observation_row["observationLevel"]= observation_levels_hash.keys[observation_levels_hash.keys.length-1]
+      end
+    rescue JSON::ParserError => e  
```

You are still ignoring this error. This is not OK.